nameko/testing/pytest.py

```diff
@@ -166,59 +164,67 @@ def destroy(vhost):
         yield vhosts
 
 
-@pytest.yield_fixture()
-def rabbit_config(request, vhost_pipeline, rabbit_manager):
+@pytest.yield_fixture
+def rabbit_uri(request, vhost_pipeline):
```

Comment:
> There is already a fixture called amqp_uri that pulls the URI out of rabbit_config. 

I left the `amqp_uri` for backward compatibility, there may be code outside using it.

> `rabbit_uri` is a bit vague because there is also the URI for the management interface

Can be `rabbit_management_uri` if needed .)

I understand the consistency worries. Made me think again. :thinking: 

With the name used, it was hard to say which is better or  more consistent. There was `rabbit_config`, `rabbit_ssl_config` and `rabbit_manager`. I added `rabbit_uri`, `rabbit_ssl_uri` and `rabbit_ssl_options`. It felt somehow inconsistent to use the protocol name for URIs and leave the broker name for the rest. Also the web module does not use the protocol name (http) for server uri and port. On the other hand `AMQP_URI` is used as config key and inside amqp extensions. Also amqp extensions _may_ in theory work with different AMQP brokers, not just with rabbit.

:thinking:

Reply:
When reasoning about it again, we've got:

- `rabbit_manager` - fits well as it wraps management API of RabbitMQ
- `rabbit_config` - combines `rabbit_uri` and `config` (used to use `rabbit_manager` via `vhost_pipeline` with implementation tying it to RabbitMQ as specific broker)
- `rabbit_uri` - now uses `rabbit_manager` via `vhost_pipeline` making it again RabbitMQ specific
- `rabbit_ssl_config`, `rabbit_ssl_uri` and `rabbit_ssl_options` - same for SSL rabbit setup

The main argument for using rabbit in `rabbit_uri` is the fact that is sets and tears down messaging setup with tooling specific to RabbitMQ message broker. Rabbit is the chosen broker (at least for testing).

Reply:
Yep, this is good reasoning. You've convinced me 😉